resources/manage.rb

```diff
@@ -33,8 +35,31 @@
   users_groups = {}
   users_groups[new_resource.group_name] = []
 
+  # If group_id is specified, create group if it doesn't exist.
+  # Prevents user block from failing if the group does not yet exist
+  # when upg = false.  The "unless" statement prevents resource
+  # conflicts to allow the group to be defined in other cookbooks in more detail.
+  unless node['etc']['group'][new_resource.group_name]
```

Comment:
Why bother with this when you have the guard clause making the exact same logical check?